### PR TITLE
Use ESM

### PR DESCRIPTION
**Pull request by wooorm**
* This commit switches the package over to actual ESM, it must be imported with `import`
* The default export is the slug class, the static function is exposed as a named export:
  ```js
  import GithubSlugger, {slug} from 'github-slugger'
  ```
* This also uses an actual ES6 class instead of prototype/constructor. This means that you now **must** use `new` to create an instance (it was optional before):
  ```js
  const slugger = new GithubSlugger()
  ```

/cc @Flet

### CODE REVIEW

File: test/test-static.js [hunk 1]
```diff
@@ -1,25 +0,0 @@
-const test = require('tape')
```
Comment (wooorm) [inline test/test-static.js:1]: Inlined this file above, and dropped some unneeded tests

File: package.json [hunk 2]
```diff
@@ -49,7 +50,7 @@
   "scripts": {
     "format": "standard --fix",
     "test-api": "tape test | tap-spec",
-    "test-coverage": "nyc --reporter lcov tape test/index.js | tap-spec",
+    "test-coverage": "c8 --check-coverage --100 --reporter lcov npm run test-api",
```
Comment (wooorm) [inline package.json:53]: `c8` uses native v8 coverage gathering so it’s faster and closer to how Node works, and it’s better at ESM

File: README.md [hunk 3]
```diff
@@ -56,7 +57,7 @@ If you need, you can also use the underlying implementation which does not keep
 track of the previously slugged strings (not recommended):
 
 ```js
-var slug = require('github-slugger').slug;
+import GithubSlugger, {slug} from 'github-slugger'
```
Comment (wooorm) [inline README.md:60]: `GithubSlugger` as a variable is unused in this example, but folks sometimes don’t know you can get the default *and* a named export, so I think it’s good to show that here

File: index.js [hunk 4]
```diff
@@ -1,50 +1,55 @@
...
-
-module.exports = BananaSlug
+import { regex } from './regex.js'
 
 const own = Object.hasOwnProperty
 
-function BananaSlug () {
-  const self = this
-
-  if (!(self instanceof BananaSlug)) return new BananaSlug()
```
Comment (wooorm) [inline index.js:10]: This isn’t possible with actual classes anymore: they must be constructed with `new`

File: index.js [hunk 5]
```diff
@@ -1,50 +1,55 @@
...
+   *
+   * @return void
+   */
+  reset () {
+    this.occurrences = Object.create(null)
+  }
 }
 
-function slugger (string, maintainCase) {
+export function slug (string, maintainCase) {
```
Comment (wooorm) [inline index.js:51]: This is now a named export instead of being a function patched on the class

**Comment by Flet** [comment]
Checking it out now!

File: index.js (hunk 4 above)
Comment (Flet) [reply index.js:10]: This code smells like old books. :)

**Review by Flet — APPROVED** [review]

**Pull request merged**